ChildProject/pipelines/metrics.py

```diff
@@ -212,10 +257,32 @@ def _process_unit(self, unit: str):
 
         metrics["wc_adu_ph"] = its["words"].sum() * 3600 / unit_duration
 
+        #add info for child_id and duration to the dataframe
         metrics["child_id"] = self.project.recordings[
             self.project.recordings[self.by] == unit
         ]["child_id"].iloc[0]
         metrics["duration"] = unit_duration
+        
+        #get and add to dataframe children.csv columns asked
+        if self.child_cols:
+            for label in self.child_cols:
+                metrics[label]=self.project.children[
+                        self.project.children["child_id"] == metrics["child_id"]
+                ][label].iloc[0]
+                
+        #get and add to dataframe recordings.csv columns asked
+        if self.rec_cols:
+            for label in self.rec_cols:
+                #for every unit drop the duplicates for that column
+                value=self.project.recordings[
+                        self.project.recordings[self.by] == unit
+                ][label].drop_duplicates()
+                #check that there is only one row remaining (ie this column has a unique value for that unit)
+                if len(value) == 1:
+                    metrics[label]=value.iloc[0]
+                #otherwise, leave the column as NA
+                else:
+                    metrics[label]="NA"
```

Comment:
oh i see! so you were already thinking about a situation like that I raised in my previous comment... I think this solution is more elegant than the one I proposed, which users may find frustrating

Reply:
Perhaps issue a warning along with it?